chalk-solve/src/recursive/solve.rs

```diff
@@ -194,9 +221,17 @@ impl<'me, I: Interner> Solver<'me, I> {
         environment: &Environment<I>,
         goal: &DomainGoal<I>,
     ) -> Result<Vec<ProgramClause<I>>, Floundered> {
-        program_clauses_for_goal(self.program, environment, goal)
+        program_clauses_for_goal(self.db(), environment, goal)
     }
 }
+
+impl<S, I> SolveIterationHelpers<I> for S
```

Comment:
Wait, extra?

Reply:
There are two traits -- one is the public method, the other is the private helpers, and you need an impl for both.